### PR TITLE
HS-1485: Allow Alerts to be triggered by multiple monitoring policies

### PR DESCRIPTION
**Pull request by vserrago**
## Description
Allows an event to generate multiple alerts. Because an alert definition is tied indirectly to a single monitoring policy via a trigger event, the alert will only ever be tied to a single monitoring policy.

Existing tests had to be updated to fix some of the following issues:
- Tests did not clear previous builder arguments.
- Tests created duplicate policies for the same tenants.

## Jira link(s)
- https://opennms.atlassian.net/browse/HS-1485

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Comment by cgorantla** [comment]
The failure in test is coming after merging my changes to this branch.

Since @vserrago updated tests, we can't use same tenant across different test steps. The test updates are extensive and it's not straight forward to fix the test with current approach.

I will come back to this PR later.

**Comment by cgorantla** [comment]
The tests are still failing.  Resolved conflicts with develop

File: alert/src/main/java/org/opennms/horizon/alertservice/service/AlertEventProcessor.java [hunk 1]
```diff
@@ -76,36 +77,59 @@ public class AlertEventProcessor {
...
         }
+
+        List<org.opennms.horizon.alertservice.db.entity.Alert> alerts = alertDefinitions.stream()
+            .map(alertDefinition -> getAlertData(event, alertDefinition))
+            .map(alertData -> createOrUpdateAlert(event, alertData))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .toList();
+
+        // FIXME: If the alert is going to be delete immediately, should we even bother creating it?
```
Comment (freddy61025) [inline alert/src/main/java/org/opennms/horizon/alertservice/service/AlertEventProcessor.java:125]: Suggest that we can create a ticket for reminding.

File: alert/src/main/java/org/opennms/horizon/alertservice/service/AlertEventProcessor.java [hunk 2]
```diff
@@ -76,36 +77,59 @@ public class AlertEventProcessor {
...
+            // No alert definition matching, no alert to create
+            eventsWithoutAlertDataCounter.increment();
+            return Collections.emptyList();
         }
+
+        List<org.opennms.horizon.alertservice.db.entity.Alert> alerts = alertDefinitions.stream()
+            .map(alertDefinition -> getAlertData(event, alertDefinition))
+            .map(alertData -> createOrUpdateAlert(event, alertData))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
```
Comment (freddy61025) [inline alert/src/main/java/org/opennms/horizon/alertservice/service/AlertEventProcessor.java:122]: The performance of a single block of map will preform a bit better unless you feel it is more readable.

**Review by freddy61025 — APPROVED** [review]
Codewise LGTM but i am not very familiar with the logics.

File: alert/src/test/java/org/opennms/horizon/alertservice/testcontainers/TestContainerRunnerClassRule.java [hunk 3]
```diff
@@ -104,6 +104,8 @@ private void startPostgresContainer() {
             .withNetworkAliases("postgres")
             .withEnv("POSTGRESS_CLIENT_PORT", String.valueOf(PostgreSQLContainer.POSTGRESQL_PORT))
             .withLogConsumer(new Slf4jLogConsumer(LOG).withPrefix("POSTGRES"));
+        // DEBUGGING: uncomment to interact/view DB
+        //postgreSQLContainer.getPortBindings().add("5432:5432");
```
[the remark this answers is not in the document]
Comment (vserrago) [reply alert/src/test/java/org/opennms/horizon/alertservice/testcontainers/TestContainerRunnerClassRule.java:108]: I intentionally left the debug port exposure commented-out, like we do with the application container

**Pull request merged**